weather_dl/download_pipeline/pipeline_test.py

```diff
@@ -135,6 +135,12 @@ def test_local_run(self):
             )
         )
 
+    def test_update_manifest(self):
```

Comment:
This is good. I'd also like to see coverage that `--update-manifest` correctly updates the manifest as we would expect. A couple ways we could do this: 
- here: let's write to a local manifest and assert that the output is correct. We also can also make multiple runs to show that the manifest handles state correctly (e.g. skips already downloaded data). 
- in a unit tests: abstract out the manifest layer and ensure that independent aspects of the feature are implemented correctly.

Reply:
Yes, definitely thats the plan -- will update the unit test.